lib/core/src/Cardano/Wallet.hs

```diff
@@ -2150,6 +2167,17 @@ guardQuit WalletDelegation{active,next} rewards = do
   where
     anyone = const True
 
+guardSelect
+    :: Coin
+    -> NonEmpty TxOut
+    -> Either ErrTxOutTooSmall ()
+guardSelect minUtxoValue txouts = do
+    let invalidTxOuts =
+            filter (\(TxOut _ out) -> out < minUtxoValue) $ NE.toList txouts
+    let getVals = map (\(TxOut _ (Coin c)) -> c)
+    unless (L.null invalidTxOuts) $
+        Left (ErrTxOutTooSmall (getCoin minUtxoValue) (getVals invalidTxOuts) )
```

Comment:
Maybe this might be easier to call? -- the ExceptT is built in.

```suggestion
guardSelect
    :: Coin
    -> NonEmpty TxOut
    -> ExceptT m ErrSelectForPayment ()
guardSelect minUtxoValue txouts
    | outputsAreAllBigEnough = pure ()
    | otherwise = throwE $ ErrSelectForPayment $ ErrTxOutTooSmall (getCoin minUtxoValue) (getVals invalidTxOuts)
  where
    outputsAreAllBigEnough = not (L.null invalidTxOuts)
    invalidTxOuts = filter (\(TxOut _ out) -> out < minUtxoValue) $ NE.toList txouts
    getVals = map (\(TxOut _ (Coin c)) -> c)
```

Reply:
probably. here I just wanted to be aligned with other guards and more fit to the style already existent.  But I will add property to guardSelect in coming commit